frontend/src/components/worksheets/Worksheet/Worksheet.js

```diff
@@ -1507,11 +1512,31 @@ class Worksheet extends React.Component {
                             this.setFocus(focus, this.state.subFocusIndex);
                         }
                     }
-                    this.setState({
-                        updating: false,
-                        version: this.state.version + 1,
-                        numOfBundles: numOfBundles,
-                    });
+                    this.setState(
+                        {
+                            updating: false,
+                            version: this.state.version + 1,
+                            numOfBundles: numOfBundles,
+                        },
+                        () => {
+                            if (addImage) {
+                                const subFocusIndex = this.state.subFocusIndex
+                                    ? this.state.subFocusIndex
+                                    : 0;
+                                let focusIndexPair = this.state.focusIndex + ',' + subFocusIndex;
+                                let index = this.state.ws.info.block_to_raw[focusIndexPair];
+                                if (index === undefined) {
+                                    // the newly uploaded image currently does not create a new block
```

Comment:
Can you update this comment to be clearer? When does "the newly uploaded image currently does not create a new block"? Does this always happen or only if `index === undefined`?

Reply:
e.g. be added as a bundle row to a table; so `does not create a new block` <==> `index === undefined`